cmd/ackdev/cmd/soak_install.go

```diff
@@ -0,0 +1,222 @@
+}
+
+var soakInstallCmd = &cobra.Command{
+	Use:     "install",
+	RunE:    soakInstall,
+	Args:    cobra.NoArgs,
+	Short:   "Installs the soak test framework onto the cluster",
+	Example: "ackdev soak install --cluster-config=./cluster-config.yml --service=s3",
+}
+
+func soakInstall(cmd *cobra.Command, args []string) (err error) {
+	if optSoakService == "" {
+		return fmt.Errorf("you must specify a service to bootstrap")
+	}
+
+	if optSoakInstallImageTag == "" {
+		return fmt.Errorf("image-tag is a required field")
+	}
+
+	if optSoakInstallImageRepository == "" {
+		optSoakInstallImageRepository, err = soak.GetRepoURL(optSoakService)
+		if err != nil {
+			return err
+		} else if optSoakInstallImageRepository == "" {
+			return fmt.Errorf("unable to find default soak image repository")
+		}
+	}
+
+	cfg, err := config.Load(ackConfigPath)
+	if err != nil {
+		return err
+	}
+
+	repoManager, err := repository.NewManager(cfg)
+	if err != nil {
+		return err
+	}
+
+	controllerRepo, err := repoManager.LoadRepository(optSoakService, repository.RepositoryTypeController)
+	if err != nil {
+		return err
+	}
+
+	testInfraRepo, err := repoManager.LoadRepository("test-infra", repository.RepositoryTypeCore)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Adding Prometheus Helm chart repository ... ")
+	soak.AddHelmRepo(defaultPrometheusRepoName, "https://prometheus-community.github.io/helm-charts")
+	fmt.Println("👍")
+
+	fmt.Printf("Installing ACK %s controller Helm chart ... ", optSoakService)
+	controllerChart := filepath.Join(controllerRepo.FullPath, "helm")
+	_, err = installController(optSoakService, controllerChart)
+	if err != nil {
+		return err
+	}
+	fmt.Println("👍")
+
+	fmt.Printf("Installing Prometheus Helm chart ... ")
+	_, err = installPrometheus(optSoakService)
+	if err != nil {
+		return err
+	}
+	fmt.Println("👍")
+
+	fmt.Printf("Applying ACK Grafana dashboard ... ")
+	grafanaKustomization := filepath.Join(testInfraRepo.FullPath, "soak/monitoring/grafana")
+	applyGrafanaDashboard(grafanaKustomization)
+	fmt.Println("👍")
+
+	fmt.Printf("Installing soak test runner Helm chart ... ")
+	soakRunnerChart := filepath.Join(testInfraRepo.FullPath, "soak/helm/ack-soak-test")
+	_, err = installSoakRunnerChart(optSoakService, soakRunnerChart, optSoakInstallImageRepository, optSoakInstallImageTag)
+	if err != nil {
+		return err
+	}
+	fmt.Println("👍")
+
+	return nil
+}
+
+func installController(service string, controllerChartPath string) (*release.Release, error) {
+	chartValues := map[string]interface{}{
+		"metrics": map[string]interface{}{
+			"service": map[string]interface{}{
+				"create": true,
+				"type":   "ClusterIP",
+			},
+		},
+		"aws": map[string]interface{}{
+			"region": defaultControllerRegion,
+		},
+		"serviceAccount": map[string]interface{}{
+			"create": false,
+			"name":   soak.GetDefaultServiceAccountName(service),
+		},
+	}
```

Comment:
this and the below for the prometheus chart have a good chance of becoming out of sync with changes made to the test-infra/soak/ values.yaml files.